nandtool/nandtool.c

```diff
@@ -193,32 +283,79 @@ static void nandtool_help(const char *prog)
 }
 
 
+static int nandtool_parseRange(char *str, int *start, int *size)
+{
+	char *tok, *endptr;
+
+	tok = strtok(str, ":");
+	if (tok == NULL) {
+		return -1;
+	}
+
+	*start = strtoul(tok, &endptr, 0);
+	if (*endptr != '\0') {
+		return -1;
+	}
+
+	tok = strtok(NULL, ":");
+	if (tok == NULL) {
+		/* Default case, one block */
+		*size = 1;
+	}
+	else {
+		*size = strtoul(tok, &endptr, 0);
+		if (*endptr != '\0') {
+			return -1;
+		}
+	}
```

Comment:
A bit simpler
```suggestion
	char *endptr;

	*start = strtoul(str, &endptr, 0);
	if (*endptr == '\0') {
		/* Default case, one block */
		*size = 1;
	}
	else if (*endptr == ':') {
		str = endptr + 1;
		*size = strtoul(str, &endptr, 0);
		if ((str == endptr) || (*endptr != '\0')) {
			return -1;
		}
	}
	else {
		return -1;
	}
```

Reply:
Thanks, that's a nice solution.

Reply:
- I think it would be nice to accept `:3`, but I should also update the help command.
- Yes, that's another advantage of the proposed solution. I'll change to `const`.